### ISSUE TITLE
'DateTime' is not a subtype of type 'TimeOfDay'

### ISSUE THREAD

**Issue by michael-golfi** [danvick/flutter_form_builder#131]
I'm seeing `'DateTime' is not a subtype of type 'TimeOfDay'` when I use a DateTimePickerFormField with time only option and clicking cancel on the time picker.

It seems that the error is happening at line 324 because currentValue is DateTime and DateTimeField.convert is expecting a TimeOfDay:

```dart
return DateTimeField.convert(await _showTimePicker(context, currentValue) ?? currentValue);
```

**Comment by michael-golfi** [comment]
I seemed to have fixed it by replacing the line with:

```dart
return DateTimeField.convert(await _showTimePicker(context, currentValue) ?? TimeOfDay.fromDateTime(currentValue));
```

**Comment by danvick** [comment]
Hi @michael-golfi,
Thanks for bringing this to my attention, the fix for this has been published in version 3.5.3.
